Add timeout attribute for send files

Konsts opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!-- 
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Add ```timeout``` param to telegram_bot.send_file(send_video, send_photo, send_sticker, send_document) that help fixed issue with ```timeout``` error (with a large file size, poor internet connection, etc.)

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box! 
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example entry for `configuration.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->

```yaml
# Example configuration.yaml
service: telegram_bot.send_video
      data_template:
        target:
         - !secret chatId
        file: "/config/video_captures/snapshotDoor.mp4"
        caption: "someone opened the door!"
        timeout: 1000
```

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #26498
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [x] Documentation added/updated for [www.home-assistant.io][docs-repository] 
Documentation PR: home-assistant/home-assistant.io#11944

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [x] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

Konsts committed:
Add timeout attribute for send files

homeassistant commented:
Hi @Konsts, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#31379).

Once you do that we will be able to review and accept this pull request.

Thanks!

springstan commented on homeassistant/components/telegram_bot/__init__.py:
```diff
@@ -153,6 +153,7 @@
         vol.Optional(ATTR_PASSWORD): cv.string,
         vol.Optional(ATTR_AUTHENTICATION): cv.string,
         vol.Optional(ATTR_VERIFY_SSL): cv.boolean,
+        vol.Optional(ATTR_TIMEOUT): cv.positive_int,
```
Isn't this a duplicate? Since you add it to the `BASE_SERVICE_SCHEMA` before and now extend it further.

Konsts replied:
Yes, you’re right, I need to make changes and create a new pull request?

springstan replied:
You should make these changes and then push them. No, you do not need to create a new pull request since this one will update automatically as soon as you push any changes.

springstan replied:
You should only add it to `BASE_SERVICE_SCHEMA` and remove this duplicate here.
Otherwise you will have to add it every single send service schema which would be redundant.

Konsts replied:
Ok

Konsts committed:
Remove dublicate

Remove attribute dublicate in BASE_SERVICE_SCHEMA

Konsts committed:
Remove duplicate

return attribute to BASE_SERVICE_SCHEMA and remove from SERVICE_SCHEMA_SEND_FILE

springstan commented:
Please add the timeout variable to `send_message` and `send_location` in services.yaml as well :)

Konsts committed:
Add timeout parameter description

Add timeout parameter description for "send message" and "send location"

springstan commented on homeassistant/components/telegram_bot/__init__.py:
```diff
@@ -135,7 +135,7 @@
         vol.Optional(ATTR_KEYBOARD): vol.All(cv.ensure_list, [cv.string]),
         vol.Optional(ATTR_KEYBOARD_INLINE): cv.ensure_list,
-        vol.Optional(CONF_TIMEOUT): vol.Coerce(float),
+        vol.Optional(ATTR_TIMEOUT): cv.positive_int,
```
@MartinHjelmare does changing the expected data type of a service count as a breaking change?

MartinHjelmare replied:
It could but here I think we're backwards compatible. We're limiting it to positive numbers though. But a negative timeout sounds weird so I think that's ok.

MartinHjelmare commented:
So nothing really changed here except the service description additions?

springstan commented:
> 
> 
> So nothing really changed here except the service description additions?

Not entirely, since the `timeout` param was added to all telegram_bot.send_file(send_video, send_photo, send_sticker, send_document) services.

MartinHjelmare commented:
I don't see that change. Where is it?

springstan commented:
Oh yeah you are right, it was only replaced. So yes you could say that only the service descriptions were added.
